src/components/TextInput/TextInput.jsx

```diff
@@ -3,61 +3,98 @@ import React from "react";
 import { classNames } from "../../utils";
 import Icon from "../Icon/Icon.jsx";
 import styles from "./styles.css";
+import { useMountEffect } from "../../hooks";
 
 const TextInput = ({
+  children,
   type,
   label,
+  placeholder,
   id,
   error,
+  success,
+  autoFocus,
   wrapperClassNames,
   inputClassNames,
+  labelClassNames,
+  messageClassNames,
   ...props
 }) => {
+  const input = React.useRef(null);
+  useMountEffect(() => autoFocus && input && input.current.focus());
+
   return (
-    <div className={classNames(styles.textinputWrapper, wrapperClassNames)}>
+    <div
+      className={classNames(
+        styles.textinputWrapper,
+        error && styles.error,
+        success && styles.success,
+        wrapperClassNames,
+        !label && styles.withoutLabel
+      )}>
       <input
         id={id}
-        placeholder=" "
-        className={classNames(
-          styles.textinput,
-          error && styles.textinputError,
-          inputClassNames
-        )}
+        ref={input}
+        placeholder={placeholder || " "}
+        className={classNames(styles.textinput, inputClassNames)}
         type={type}
         {...props}
       />
-      <label htmlFor={id} className={styles.textinputLabel}>
-        {label}
-      </label>
-      <Icon
-        type="alert"
-        backgroundColor="#ed6d47"
-        iconColor="#feb8a5"
-        className={classNames(
-          styles.errorIcon,
-          error && styles.errorIconActive
+      {label && (
+        <label
+          htmlFor={id}
+          className={classNames(
+            styles.textinputLabel,
+            placeholder && styles.staticTextInputLabel,
+            labelClassNames
+          )}>
+          {label}
+        </label>
+      )}
+      <div>
+        {children && <div className={styles.childrenWrapper}>{children}</div>}
+        {error && (
+          <Icon
+            type="alert"
+            backgroundColor="#ed6d47"
```

Comment:
Does these colors working properly in dark theme mode ?

Reply:
@amassarwi Thanks for the review. I did not change these colors. They are theme-independent and look great in dark theme mode: 
<img width="316" alt="DeepinScreenshot_select-area_20210525184139" src="https://user-images.githubusercontent.com/52497040/119536491-6ce44300-bd89-11eb-8557-a7deedfb2ac9.png">